src/Core/Services/IFeatureRoutedCacheService.cs

```diff
@@ -0,0 +1,19 @@
+﻿using Bit.Core.AdminConsole.Entities;
+using Bit.Core.AdminConsole.Entities.Provider;
+using Bit.Core.AdminConsole.Models.Data.Provider;
+using Bit.Core.Models.Data.Organizations;
+
+namespace Bit.Core.Services;
+
+public interface IFeatureRoutedCacheService
```

Comment:
I would love some feedback on the class name. This seems like something we should keep long-term, as the problem of feature flagging a singleton service will likely come up again. 

That said, let's not spend too much time on this; something reasonable is fine. There will be a cleanup task to remove the feature flag later, so we can spend more time on it then if needed.